### PR TITLE
[KYUUBI #2331] Add createSession method to further abstract openSession

### LINKED ISSUE #2331: [Improvement] Add createSession method to further abstract openSession
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Add createSession method to further abstract openSession

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

close #2331

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by wForget** [comment]
@yaooqinn @ulysses-you @pan3793 could you help me review this?

**Comment by ulysses-you** [comment]
what's the semantics of `createSession`, I can not see a clearly definition between `createSession` and `openSession`. e.g. Spark engine, you have already new a spark session at `createSession` phase. Shall we need move them to `openSession` ?

**Comment by wForget** [comment]
> what's the semantics of `createSession`, I can not see a clearly definition between `createSession` and `openSession`. e.g. Spark engine, you have already new a spark session at `createSession` phase. Shall we need move them to `openSession` ?

This is exactly the same as the previous semantics, `createSession` is just to create a session object for `openSession`.

**Comment by wForget** [comment]
We can make the `createSession` method protected.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 1]
```diff
@@ -51,12 +51,12 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
...
-  override def openSession(
+  override def createSession(
       protocol: TProtocolVersion,
       user: String,
       password: String,
       ipAddress: String,
-      conf: Map[String, String]): SessionHandle = {
+      conf: Map[String, String]): Session = {
     val clientIp = conf.getOrElse(CLIENT_IP_KEY, ipAddress)
     info(s"Opening session for $user@$clientIp")
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:61]: shall we move this to `openSession` ?

Comment (wForget) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:61]: > shall we move this to `openSession` ?

yeah, I will do that.

**Comment by yaooqinn** [comment]
> During the implementation of https://github.com/apache/incubator-kyuubi/issues/2301, I want to add limiter in openSession of SessionManager, So I want to implement openSession method in SessionManager add createSession abstract method.

What is the magic of breaking it into 2 phases? I don't quite follow

**Comment by wForget** [comment]
> What is the magic of breaking it into 2 phases? I don't quite follow

For this pr, it is just an improvement to the previous code, without adding any functional modifications.
I just want to make the responsibilities of the two phases look more single.

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**